concent_api/core/payments/storage.py

```diff
@@ -0,0 +1,152 @@
+import logging
+from typing import Dict
+from typing import List
+from hexbytes import HexBytes
+from ethereum.transactions import Transaction
+from golem_sci.transactionsstorage import TransactionsStorage
+
+from django.db import transaction
+
+from core.models import GlobalTransactionState
+from core.models import PendingEthereumTransaction
+
+from core.exceptions import TransactionNonceNotMatchError
+from common.constants import ErrorCode
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseTransactionsStorage(TransactionsStorage):
+    """
+    Concent custom implementation of TransactionsStorage interface used to store Ethereum transaction data into
+    database using Django models.
+    """
+
+    def __init__(self, nonce: int, *args: List, **kwargs: Dict) -> None:
+        super().__init__(*args, **kwargs)
+
+        assert isinstance(nonce, int)
+        if not GlobalTransactionState.objects.filter(pk=0).exists():
+            global_transaction_state = GlobalTransactionState(
+                pk=0,
+                nonce=nonce,
+            )
+            global_transaction_state.full_clean()
+            global_transaction_state.save()
+
+    @transaction.atomic(using='control')
+    def get_nonce(self) -> int:
+        """
+        Returns the nonce for the next transaction.
+        """
+        global_transaction_state = self._get_locked_global_transaction_state()
+        return int(global_transaction_state.nonce)
```

Comment:
Add method to `GlobalTransactionState` called `get_nonce` which will change it to int and reuse it in all places.

Reply:
Done